client/libra-dev/include/data.h

```diff
@@ -0,0 +1,181 @@
+ * @param[out] len is the length of the signed transaction memory buffer.
+*/
+enum LibraStatus libra_SignedTransactionBytes_from(const uint8_t sender_private_key[32], const uint8_t receiver[32], uint64_t sequence, uint64_t num_coins, uint64_t max_gas_amount, uint64_t gas_unit_price, uint64_t expiration_time_secs, uint8_t** ptr_buf, size_t* ptr_len);
+
+/*!
+ * Function to free the allocation memory in rust for bytes
+ * @param buf is the pointer to the bytes allocated in rust, and needs to be freed from client side
+ */
+void libra_free_bytes_buffer(const uint8_t* buf);
+
+/*!
+ * Decode LibraSignedTransaction from bytes in SignedTransaction proto.
+ *
+ * @param[in] buf contains encoded bytes of txn_bytes
+ * @param[in] len is the length of the signed transaction memory buffer.
+ * @param[out] caller allocated LibraSignedTransaction to write into.
+ *
+ * @returns status code, one of LibraAPIStatus
+*/
+enum LibraStatus libra_LibraSignedTransaction_from(const uint8_t *buf, size_t len, struct LibraSignedTransaction *out);
+
+/*!
+ * Get serialized raw transaction from a list of transaction parameters
+ *
+ * To get the serialized raw transaction in a memory safe manner, the client needs to pass in a pointer to a pointer to the allocated memory in rust
+ * and call free on the memory address with `libra_free_bytes_buffer`.
+ * @param[out] buf is the pointer that will be filled with the memory address of the transaction allocated in rust. User takes ownership of pointer returned by *buf, which needs to be freed using libra_free_bytes_buffer
+ * @param[out] len is the length of the raw transaction memory buffer.
+*/
+enum LibraStatus libra_RawTransactionBytes_from(const uint8_t sender[32], const uint8_t receiver[32], uint64_t sequence, uint64_t num_coins, uint64_t max_gas_amount, uint64_t gas_unit_price, uint64_t expiration_time_secs, uint8_t** buf, size_t* len);
+
+/*!
+ * This function takes in a raw transaction, public key and signature in bytes, and return a signed transaction in bytes.
+ * To get the serialized signed transaction in a memory safe manner, the client needs to pass in a pointer to a pointer to the allocated memory in rust
+ * and call free on the memory address with `libra_free_bytes_buffer`.
+ * @param[out] buf_result is the pointer that will be filled with the memory address of the transaction allocated in rust. User takes ownership of pointer returned by *buf, which needs to be freed using libra_free_bytes_buffer
+ * @param[out] len_result is the length of the signed transaction memory buffer.
+*/
+enum LibraStatus libra_RawTransaction_sign(const uint8_t *buf_raw_txn, size_t len_raw_txn, const uint8_t *buf_public_key, size_t len_public_key, const uint8_t *buf_signature, size_t len_signature, uint8_t** buf_result, size_t* len_result);
```

Comment:
Are public keys variable length here? Why are private keys fixed size and public keys not?